modules/linear_boltzmann_solvers/lbs_solver/io/flux_moments_io.cc

```diff
@@ -164,86 +134,80 @@ LBSSolverIO::ReadFluxMoments(LBSSolver& lbs_solver,
 
-  // Read cell nodal locations
+  // Read in mesh information
+  const auto file_cell_ids = H5ReadDataset1D<uint64_t>(file_id, "mesh/cell_ids");
+  const auto file_num_cell_nodes = H5ReadDataset1D<uint64_t>(file_id, "mesh/num_cell_nodes");
+
+  const auto nodes_x = H5ReadDataset1D<double>(file_id, "mesh/nodes_x");
+  const auto nodes_y = H5ReadDataset1D<double>(file_id, "mesh/nodes_y");
+  const auto nodes_z = H5ReadDataset1D<double>(file_id, "mesh/nodes_z");
+
+  // Validate mesh compatibility
+  uint64_t curr_node = 0;
   std::map<uint64_t, std::map<uint64_t, uint64_t>> file_cell_nodal_mapping;
   for (uint64_t c = 0; c < file_num_local_cells; ++c)
   {
-    // Read cell-id and num_nodes
-    uint64_t file_cell_global_id;
-    uint64_t file_num_cell_nodes;
+    const uint64_t cell_global_id = file_cell_ids[c];
+    const auto& cell = grid.cells[cell_global_id];
 
-    file.read((char*)&file_cell_global_id, sizeof(uint64_t));
-    file.read((char*)&file_num_cell_nodes, sizeof(uint64_t));
-
-    // Read node locations
-    std::vector<Vector3> file_nodes;
-    file_nodes.reserve(file_num_cell_nodes);
-    for (uint64_t i = 0; i < file_num_cell_nodes; ++i)
-    {
-      double x, y, z;
-      file.read((char*)&x, sizeof(double));
-      file.read((char*)&y, sizeof(double));
-      file.read((char*)&z, sizeof(double));
-
-      file_nodes.emplace_back(x, y, z);
-    } // for file node i
-
-    if (not grid.IsCellLocal(file_cell_global_id))
+    if (not grid.IsCellLocal(cell_global_id))
       continue;
 
-    const auto& cell = grid.cells[file_cell_global_id];
-
     // Check for cell compatibility
     const auto nodes = discretization.GetCellNodeLocations(cell);
 
-    OpenSnLogicalErrorIf(nodes.size() != file_num_cell_nodes,
+    OpenSnLogicalErrorIf(nodes.size() != file_num_cell_nodes[c],
                          "Incompatible number of cell nodes encountered on cell " +
-                           std::to_string(file_cell_global_id) + ".");
+                           std::to_string(cell_global_id) + ".");
+
+    std::vector<Vector3> file_nodes;
+    file_nodes.reserve(file_num_cell_nodes[c]);
+    for (uint64_t n = 0; n < file_num_cell_nodes[c]; ++n)
+    {
+      file_nodes.emplace_back(nodes_x[curr_node], nodes_y[curr_node], nodes_z[curr_node]);
+      ++curr_node;
+    }
 
     // Map the system nodes to file nodes
-    bool mapping_successful = true; // true until disproven
-    auto& mapping = file_cell_nodal_mapping[file_cell_global_id];
-    for (uint64_t n = 0; n < file_num_cell_nodes; ++n)
+    auto& mapping = file_cell_nodal_mapping[cell_global_id];
+    for (uint64_t n = 0; n < file_num_cell_nodes[c]; ++n)
     {
       bool mapping_found = false;
+      std::cout << std::endl;
       for (uint64_t m = 0; m < nodes.size(); ++m)
-        if ((nodes[m] - file_nodes[n]).NormSquare() < 1.0e-12)
+        if ((nodes[m] - file_nodes[n]).NormSquare() < 1.0e-10)
```

Comment:
Was there a reason to change the tolerance?

Reply:
I had changed it as part of debugging. I have reverted to the original